src/accelerate/utils/modeling.py

```diff
@@ -309,8 +309,13 @@ def set_module_tensor_to_device(
             device = "cpu"
         if value is None:
             new_value = old_value.to(device)
-            if dtype is not None and device in ["meta", torch.device("meta")]:
-                new_value = new_value.to(dtype)
+            if device in ["meta", torch.device("meta")]:
+                if dtype is None:
+                    # For compatibility with PyTorch load_state_dict which converts state dict dtype to existing dtype in model
+                    new_value = new_value.to(old_value.dtype)
```

Comment:
not sure if this one is needed - it's the `elif` below that solves the issue I have

Reply:
Indeed you can remove that line as it does nothing. new value is just old_value moved to `device`.

Reply:
sound good - will update tomorrow!